velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -178,21 +177,20 @@ struct UnixTimestampParseFunction {
   void setTimezone(const core::QueryConfig& config) {
     auto sessionTzName = config.sessionTimezone();
     if (!sessionTzName.empty()) {
-      sessionTzID_ = tz::getTimeZoneID(sessionTzName);
+      sessionTimeZone_ = tz::locateZone(sessionTzName);
     }
   }
 
-  int16_t getTimezoneId(const DateTimeResult& result) {
-    // If timezone was not parsed, fallback to the session timezone. If there's
-    // no session timezone, fallback to 0 (GMT).
-    return result.timezoneId != -1 ? result.timezoneId
-                                   : sessionTzID_.value_or(0);
+  const tz::TimeZone* getTimeZone(const DateTimeResult& result) {
+    // If timezone was not parsed, fallback to the session timezone.
+    return result.timezoneId != -1 ? tz::locateZone(result.timezoneId)
```

Comment:
Do we plan to get ride of DateTimeReulst::timezoneId next?

Reply:
yes. I looked into making the data time parser return the time zone pointer directly, but that will be a larger change. Will look into that next.